### DISCUSSION TITLE
Error: IO Failure to open post.force file

### DISCUSSION THREAD

**Discussion by dannypanduro** [neperfepx/FEPX#9] [Q&A] [answered]
Hi, I am having trouble getting FEPX to run on some simple models I made through Neper. I have copied a simulation file from an example, edited it to what I needed, and then moved it into the directory with my Neper model. (Photo of simulation file below)
![image](https://user-images.githubusercontent.com/81585331/112884540-78b2d200-9084-11eb-8352-fc5386c10007.png)

I am running this simulation on a simple 10-grain model I made in neper using the following commands.
$ neper -T -n 10 -oricrysym hexagonal
$ neper -M n10-id1.tess -elttype hex -rcl 0.8
My Neper is version 4.1.2
![image](https://user-images.githubusercontent.com/81585331/112885916-22df2980-9086-11eb-96cb-5751af462de9.png)

I then renamed the mesh file to 'simulation.msh' so that FEPX would run it. I have unbuntu open and made sure I was in the correct file directory. Then typed 'fepx' to run the simulation file.

I ended up with an Error: 'IO Failure to open post.force.' I also notice that the Elt number is 0 when the mesh file contains 2197 elements.
![fepx_code_no_work](https://user-images.githubusercontent.com/81585331/112885825-0cd16900-9086-11eb-837d-953abf082cee.png)

The program did write the output files I requested, but they're completely empty.
![image](https://user-images.githubusercontent.com/81585331/112886100-55892200-9086-11eb-94b1-232456765ecc.png)

I am not familiar with using Linux or many softwares, so I feel like I might be making a beginner mistake somewhere? 

I appreciate any help that is offered, thank you!

Richard Panduro

**Comment by mkasemer** [comment, marked as answer]
FEPX supports only ten node tetrahedral elements, per the documentation. When meshing with neper, you cannot use hex elements (`-elttype hex`), and you must specify the order of the elements to be 2, or quadratic (`-order 2`). Look at the mesh generation shell scripts provided with the examples (`generate_mesh.sh`) to view standard tessellation and meshing techniques. For example, the "1_uniaxial" example provided with the code:

```
#!/bin/bash

# Generate the tessellation and mesh files for use in example `1_uniaxial'
# - This bash script requires a configured installation of Neper 4.0.0
#   in order to be properly executed.

# First, generate a centroidal tessellation for the domain with `Neper -T':
neper -T -n 100 -reg 1 -rsel 1.25 -mloop 4 \
    -morpho "diameq:1,1-sphericity:lognormal(0.145,0.03)" -morphooptistop val=5e-3 \
    -oricrysym "cubic" \
    -o simulation

# Then, generate a coarse finite element mesh for the domain with `Neper -M':
neper -M simulation.tess -order 2 -rcl 1.25 -part 2

exit 0

# This script produces output files:
# - simuation.tess
# - simulation.msh
```

Note as well that you have to be careful with partitioning to make sure your simulation runs efficiently. You should partition to the number of processors you plan on running the simulation on.

Please remesh, and let us know if you still receive an error!

Best,

Matt

**Reply by dannypanduro** [reply]
It ran!! It was definitely the elements. Wow, thank you so much! I appreciate the help and tips immensely! 

Sincerly,

Richard Panduro
